### PR TITLE
Update Python docstrings in core models for better code formatting in model reference

### LINKED ISSUE #12435: Documentation - Clean up the *Reference > Pages* sub-pages to be only about Page models
### Pertinent section of the Wagtail docs

> https://docs.wagtail.org/en/stable/reference/pages/index.html
Image

### Details

* Within the `Reference` section of our documentation, we have a sub-page called `Pages`. This contains the 'Model reference' and 'Panel types' documentation, however, these are not specific to the Page models.
* I have found it confusing but have kind of gotten used to it, but from a newcomer perspective it feels odd. Panels are now fully supported across Snippets and Pages, the Model reference page has all models (not just pages).
* There are a few other changes that would make this 'Reference > Pages' section a bit more consistent and predictable.

#### Proposed changes

1. Move [`Panel types`](https://docs.wagtail.org/en/stable/reference/pages/panels.html) directly under [`Reference`](https://docs.wagtail.org/en/stable/reference/index.html).
2. Move [`Model Reference`](https://docs.wagtail.org/en/stable/reference/pages/model_reference.html) directly under [`Reference`](https://docs.wagtail.org/en/stable/reference/index.html).
3. Update the title of [`Model Reference`](https://docs.wagtail.org/en/stable/reference/pages/model_reference.html) to `Model reference` (lower case `r`, as this is the way most page titles are capitalised and `Reference` is not a proper noun.
4. Split the **tagging** section from [`Recipes`](https://docs.wagtail.org/en/stable/reference/pages/model_recipes.html) out to a new page called Tags to directly under `Advanced topics` called `Tags`.
5. Fix the incorrect heading level nesting in [`Recipes`](https://docs.wagtail.org/en/stable/reference/pages/model_recipes.html) for the 'adding endpoints with custom route methods, it should be the same level as 'overriding the serve method' not one under it.
6. Minor update to the text `wagtail.models` to use double backticks so that it appears highlighted as code. See https://github.com/wagtail/wagtail/blob/105338d0d63c5a8adf285d3c077a922f53a15d02/wagtail/models/__init__.py#L2 (we may also want to fix up the other backtick usage in that docstring to be double backticks)  * See https://github.com/wagtail/wagtail/blob/105338d0d63c5a8adf285d3c077a922f53a15d02/wagtail/models/__init__.py#L331C16-L331C31 for an example of the double backticks being used correctly to style as pre/code in the published docs. 

### Working on this

* Feel free to cover items 1-3 as a stand-alone PR, it may make reviews easier.
* Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

### PR DESCRIPTION
**Pull request by Srishti-j18**
- Relates to #12435 
- Used double backticks everywhere on `wagtail/wagtail/models/_init_.py` so that text appears highlighted as code.
- Update formatting and spelling to align better with documentation style guide
- See https://docs.wagtail.org/en/latest/reference/models.html 

## Example

Example issue `wagtail.models` is not formatted & `wagtail.models.sites import Site` is shown in italics, both of these should be treated as inline (pre) code blocks.

Screenshot 2024-10-28 at 8 08 55 AM

### CODE REVIEW

**Comment by thibaudcolas** [comment]
@Srishti-j18 thank you for giving this a go. There are a lot more changes here than your PR description suggests. Please take a moment to review your own submission before opening a pull request, or mark it as draft if you’re not sure.

**Comment by Srishti-j18** [comment]
Hi @thibaudcolas,

I raised two pull requests for this issue and completed the other five tasks in that separate pull request (https://github.com/wagtail/wagtail/pull/12437).

**Comment by laymonage** [comment]
> I raised two pull requests for this issue and completed the other five tasks in that separate pull request (#12437). I created a separate pull request for the sixth task because it may require some changes if needed.

If that's the case, then the branches should be completely separate, i.e. the branch for this PR should not contain the changes for that other PR. It looks like the changes in this PR can be applied separately from that other one, so it should be possible to make this branch independent.

**Comment by Srishti-j18** [comment]
> > I raised two pull requests for this issue and completed the other five tasks in that separate pull request (#12437). I created a separate pull request for the sixth task because it may require some changes if needed.
> 
> If that's the case, then the branches should be completely separate, i.e. the branch for this PR should not contain the changes for that other PR. It looks like the changes in this PR can be applied separately from that other one, so it should be possible to make this branch independent.

My bad, I have removed the changes from this PR that overlapped with the other PR.
Kindly review both my PRs: [#12437](https://github.com/wagtail/wagtail/pull/12437) and [#12438](https://github.com/wagtail/wagtail/pull/12438) for the issue: [#12435](https://github.com/wagtail/wagtail/issues/12435).
Apologies for the confusion.

**Review by lb- — CHANGES_REQUESTED** [review]
@Srishti-j18  thanks for this, I will rebase and merge with a few fixes.

Please note that the format for the Python DocStrings is RST, and adding double backticks is important for formatting values as `pre` text.

however, your PR included a lot of additional double backticks for things such as 

```
:meth:``
:attr:``
```

This is incorrect, and will actually break the formatting.

I know it's a bit confusing compared to Markdown, but in the future be sure to build your docs locally and check your changes before preparing a PR.

Thank you though for getting this updated, I will clean up and merge shortly.

File: wagtail/models/__init__.py [hunk 1]
```diff
@@ -2711,7 +2711,7 @@ def serve_password_required_response(self, request, form, action_url):
 
             if hasattr(settings, "PASSWORD_REQUIRED_TEMPLATE"):
                 warn(
-                    "The `PASSWORD_REQUIRED_TEMPLATE` setting is deprecated - use `WAGTAIL_PASSWORD_REQUIRED_TEMPLATE` instead.",
+                    "The ``PASSWORD_REQUIRED_TEMPLATE`` setting is deprecated - use ``WAGTAIL_PASSWORD_REQUIRED_TEMPLATE`` instead.",
```
Comment (lb-) [inline wagtail/models/__init__.py:2714]: No need to add double backticks to errors/warnings, this goes straight to stdout/console and is not needing to be formatted.

**Review by lb- — APPROVED** [review]
Thanks for getting this started @Srishti-j18  - looking much better now.

Screenshot 2024-10-28 at 8 21 08 AM

**Comment by Srishti-j18** [comment]
> @Srishti-j18 thanks for this, I will rebase and merge with a few fixes.
> 
> Please note that the format for the Python DocStrings is RST, and adding double backticks is important for formatting values as `pre` text.
> 
> however, your PR included a lot of additional double backticks for things such as
> 
> ```
> :meth:``
> :attr:``
> ```
> 
> This is incorrect, and will actually break the formatting.
> 
> I know it's a bit confusing compared to Markdown, but in the future be sure to build your docs locally and check your changes before preparing a PR.
> 
> Thank you though for getting this updated, I will clean up and merge shortly.

Apologies for not checking it before making the PR. In the future, I'll keep that in mind.

**Pull request merged**